src/IO/S3/PocoHTTPClient.cpp

```diff
@@ -314,30 +291,81 @@ void PocoHTTPClient::addMetric(const Aws::Http::HttpRequest & request, S3MetricT
         ProfileEvents::increment(disk_s3_events_map[static_cast<unsigned int>(type)][static_cast<unsigned int>(kind)], amount);
 }
 
-void PocoHTTPClient::addLatency(const Aws::Http::HttpRequest & request, S3LatencyType type, LatencyBuckets::Count amount) const
+void PocoHTTPClient::observeLatency(const Aws::Http::HttpRequest & request, S3LatencyType type, Histogram::Value latency) const
 {
-    if (amount == 0)
+    if (latency == 0)
         return;
 
-    static const LatencyBuckets::Event events_map[static_cast<size_t>(S3LatencyType::EnumSize)][static_cast<size_t>(S3MetricKind::EnumSize)] = {
-        {LatencyBuckets::S3FirstByteReadAttempt1Microseconds, LatencyBuckets::S3FirstByteWriteAttempt1Microseconds},
-        {LatencyBuckets::S3FirstByteReadAttempt2Microseconds, LatencyBuckets::S3FirstByteWriteAttempt2Microseconds},
-        {LatencyBuckets::S3FirstByteReadAttemptNMicroseconds, LatencyBuckets::S3FirstByteWriteAttemptNMicroseconds},
-        {LatencyBuckets::S3ConnectMicroseconds, LatencyBuckets::S3ConnectMicroseconds},
-    };
+    if (type == S3LatencyType::Connect)
+    {
+        const Histogram::Buckets connect_buckets = {100, 1000, 10000, 100000, 200000, 300000, 500000, 1000000, 1500000};
```

Comment:
Should these buckets be made configurable so that it would be easier to tune if it's a config? Also, worth leaving a comment how we came up with these intervals if we are going to go with fix bucket intervals.

Reply:
> Also, worth leaving a comment how we came up with these intervals if we are going to go with fix bucket intervals.

I just copied and pasted them from the previous values.

> Should these buckets be made configurable so that it would be easier to tune if it's a config?

Good suggestion, I'll raise a separate PR for this.